src/dolphin/interpolation.py

```diff
@@ -0,0 +1,149 @@
+            weights[i,j] = False if radar pixel (i,j) is not a PS
+        Can also pass a coherence image to use as weights.
+    num_neighbors: int (optional)
+        number of nearest PS pixels used for interpolation
+        num_neighbors = 20 by default
+    max_radius : int (optional)
+        maximum radius (in pixels) for PS searching
+        max_radius = 51 by default
+    min_radius : int (optional)
+        minimum radius (in pixels) for PS searching
+        max_radius = 0 by default
+    alpha : float (optional)
+        hyperparameter controlling the weight of PS in interpolation: smaller
+        alpha means more weight is assigned to PS closer to the center pixel.
+        alpha = 0.75 by default
+    weight_cutoff: float
+        Threshold to use on `weights` so that pixels where
+        `weight[i, j] < weight_cutoff` have phase values replaced by
+        an interpolated value.
+        If `weight_cutoff = 0` (default),  All pixels are replaced with a
+        smoothed version of the surrounding pixels.
+        If `weight_cutoff = 1`, only pixels with exactly weight=1
+        are kept, and the rest are replaced with an interpolated value.
+
+    Returns
+    -------
+    interpolated_ifg : 2D complex array
+        interpolated interferogram with the same amplitude, but different
+        wrapped phase at non-ps pixels.
+
+    References
+    ----------
+    "A persistent scatterer interpolation for retrieving accurate ground
+    deformation over InSAR\-decorrelated agricultural fields"
+    Chen et al., 2015, https://doi.org/10.1002/2015GL065031
+
+    """
+    nrow, ncol = weights.shape
+
+    if np.all(
+        np.logical_or(
+            np.logical_or(weights == 0, weights == 1),
+            np.logical_or(weights is True, weights is False),
+        )
+    ):
+        logger.info("Binary weights, using PS-like interpolation.")
+    else:
+        logger.info(
+            f"Range of values as weights, using weight_cutoff = {weight_cutoff}"
+        )
+
+    # Ensure weights are between 0 and 1
+    if np.any(weights.astype(np.float32) > 1):
+        logger.warning("weights array has values greater than 1. Clipping to 1.")
+    if np.any(weights.astype(np.float32) < 0):
+        logger.warning("weights array has negative values. Clipping to 0.")
+    # Make shared versions of the input arrays to avoid copying in each thread
+    weights_float = np.clip(weights.astype(np.float32), 0, 1)
+
+    interpolated_ifg = np.zeros((nrow, ncol), dtype=np.complex64)
+
+    indices = np.array(get_circle_idxs(max_radius, min_radius=min_radius))
+
+    _interp_loop(
+        ifg,
+        weights_float,
+        weight_cutoff,
+        num_neighbors,
+        alpha,
+        indices,
+        interpolated_ifg,
+    )
+    return interpolated_ifg
+
+
+@numba.njit(parallel=True)
+def _interp_loop(
+    ifg, weights, weight_cutoff, num_neighbors, alpha, indices, interpolated_ifg
+):
```

Comment:
I wonder if you could add a simple unit test for this function. for example if we assume a simple phase ramp with a known model coefficients and randomly mask out pixels as invalid pixels, and interpolate the phase of valid pixels, then I expect the interpolated phase over invalid pixels to be close to the value we get from the model coefficients with a tolerance.

Reply:
@mirzaees I don't see this unit test added? Are you planning to add or have a different plan?